app/lib/frontend/search_service.dart

```diff
@@ -85,14 +86,27 @@ Future<SearchResultPage> _loadResultForPackages(SearchQuery query,
   final List<Key> devVersionKeys =
       packageEntries.map((p) => p.latestDevVersionKey).toList();
   if (versionKeys.isNotEmpty) {
+    // Analysis data fetched concurrently to reduce overall latency.
+    final Future<List<AnalysisData>> analysisBatchFuture = Future.wait(
+        packageEntries.map(
+            (p) => analyzerClient.getAnalysisData(p.name, p.latestVersion)));
+
     final allVersions =
         await dbService.lookup([]..addAll(versionKeys)..addAll(devVersionKeys));
     final versions = allVersions.sublist(0, versionKeys.length);
     final devVersions = allVersions.sublist(versionKeys.length);
-    return new SearchResultPage(
-        query, totalCount, versions, devVersions, backend);
+
+    final List<AnalysisData> analysisDataList = await analysisBatchFuture;
```

Comment:
We have to be quite careful here:
```dart
Future a = ...;
Future b = ...;
await b;
await a;
```
If during `await b` `a` completes with an error, there will be no handler registered for `a`. This will cause the error to be propagated to the current zone.

So usually it's better to do:

```dart
Future a =  ...;
// ... only synchronous code here
Future b = ...; 
// ... only synchronous code here
await Future.wait([a, b]);
```
Then one avoids propagating an error to the zone, since it's guaranteed that there is a registered callback on both futures, which will make `_loadResultForPackages` return with that error.

Reply:
good point, fix ongoing